File: ruby/raindrops/raindrops.rb
class Raindrops
  TRANSLATIONS = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }
  private_constant :TRANSLATIONS

  def self.convert(count)
    message = ''

    TRANSLATIONS.each do |(num, word)|
      message << word if count % num == 0
    end

    message.empty? ? count.to_s : message
  end
end

File: clojure/raindrops/raindrops.clj
(ns raindrops)

(def ^:private translations
  {3 "Pling"
   5 "Plang"
   7 "Plong"})

(defn- translate [phrase]
  (reduce-kv (fn [translation number word]
              (if (zero? (mod phrase number))
                (str translation word)
                translation))
             ""
             translations))

(defn convert [phrase]
  (let [translation (translate phrase)]
    (if (empty? translation)
      (str phrase)
      translation)))
